app.py

```diff
@@ -0,0 +1,143 @@
+from mmif import Mmif, View, Annotation, Document, AnnotationTypes, DocumentTypes
+import pickle
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.feature_extraction.text import TfidfTransformer
+
+# For an NLP tool we need to import the LAPPS vocabulary items
+from lapps.discriminators import Uri
+
+# Import functions from tfidf.py
+from tfidf import sort_coo, extract_topn_from_vector, get_keywords
+
+class TfidfKeywordextractor(ClamsApp):
+
+    def __init__(self):
+        super().__init__()
+
+    def _appmetadata(self):
+        # see https://sdk.clams.ai/autodoc/clams.app.html#clams.app.ClamsApp._load_appmetadata
+        # Also check out ``metadata.py`` in this directory.
+        # When using the ``metadata.py`` leave this do-nothing "pass" method here.
+        pass
+
+    def _annotate(self, mmif: Mmif, **parameters) -> Mmif:
+        # see https://sdk.clams.ai/autodoc/clams.app.html#clams.app.ClamsApp._annotate
+        for arg, val in parameters.items():
+            print("Parameter %s=%s" % (arg, val))
+            # as we defined this `error` parameter in the app metadata
+            if arg == 'error' and val is True:
+                raise Exception("Exception - %s" % parameters['error'])
+
+        # Initialize the MMIF object from the string if needed
+        self.mmif = mmif if type(mmif) is Mmif else Mmif(mmif)
+
+        # process the text documents in the documents list
+        for doc in self.mmif.get_documents_by_type(DocumentTypes.TextDocument):
+            idf_file = parameters['idf_file']
+            feature_dict_file = parameters['feature_dict_file']
+            topn = parameters['topn']
+            new_view = self._new_view(parameters, doc.id)
+            # _run_nlp_tool() is the method that does the actual work
+            self._run_nlp_tool(doc, new_view, doc.id, idf_file, feature_dict_file, topn)
+        # return the MMIF object
+        return self.mmif
+
+    def _new_view(self, runtime_config, docid=None):
+        view = self.mmif.new_view()
+        view.metadata.app = self.metadata.identifier
+        # first thing you need to do after creating a new view is "sign" the view
+        # the sign_view() method will record the app's identifier and the timestamp
+        # as well as the user parameter inputs. This is important for reproducibility.
+        self.sign_view(view, runtime_config)
+        # then record what annotations you want to create in this view
+        # view.new_contain(Uri.DOCUMENT, document=docid)
+        view.new_contain(AnnotationTypes.Alignment)
+        return view
+
+    def _run_nlp_tool(self, doc, new_view, full_doc_id, idf_file, feature_dict_file, topn):
+        """Run the NLP tool over the document and add annotations to the view, using the
+        full document identifier (which may include a view identifier) for the document
+        property."""
+        text = doc.text_value
+
+        # load idf values
```

Comment:
can't these to pickle files merged into a single pickle? I think that will be help eliminating one of the parameter and hence reduce a source of mis-configuration 

BTW do these two "model" files actually work independently? i.e., can you use `feature_dict` from round X of training and`idf_vaules`  from training round Y together? If that's not the case, having two parameters for these doesn't make sense to start with, IMHO.